File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/NoArgsCommand.java
package ro.razvan.java.dagger.atm;

import java.util.List;

public interface NoArgsCommand extends Command {

    @Override
    default Result handleInput(List<String> input) {
        if (!input.isEmpty()) return Result.invalid();
        return handle();
    }

    Result handle();
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/NoArgsCommand.kt
package ro.razvan.kotlin.dagger.atm

import ro.razvan.kotlin.dagger.atm.Command.Result

interface NoArgsCommand : Command {

    override fun handleInput(input: List<String>): Result {
        if (input.isNotEmpty()) {
            return Result.invalid()
        }
        return handle()
    }

    fun handle(): Result

}